### ISSUE TITLE
How to handle the different number of positive and negative samples?

### ISSUE THREAD

**Issue by xmengli** [xiangli13/circle-loss#2]
Thanks for sharing the code!

I noticed that the implementation supports the calculation between 1 positive pair and 1 negative pair.  
If I have 1 positive pair and N negative pair, I am wondering how to handle the loss more effectively?

Thanks!

**Comment by haohang96** [comment]
Even if there is only 1 positive pair and 1 negtive pair, I think the loss computed by binary_crossentropy in this code is not same as original circle loss:
let's use $lp$ denote positive logits, $ln$ denote negtive logits
original circle loss: L = log(1 + e^(ln)*e^(-lp)) # eq1
loss of this code: L = -log(1/(1+e^(-lp))) + -log(e^(-ln)/(1+e^(-ln))) 
                               = log(1+e^(-lp))*(1+e^(ln)) # eq2


I think #eq1 and #eq2 are not same

**Comment by haohang96** [comment]
> Thanks for sharing the code!
> 
> I noticed that the implementation supports the calculation between 1 positive pair and 1 negative pair.
> If I have 1 positive pair and N negative pair, I am wondering how to handle the loss more effectively?
> 
> Thanks!

Even if there is only 1 positive pair and 1 negtive pair, I think the loss computed by binary_crossentropy in this code is not same as original circle loss:
let's use $lp$ denote positive logits, $ln$ denote negtive logits
original circle loss: L = log(1 + e^(ln)e^(-lp)) # eq1
loss of this code: L = -log(1/(1+e^(-lp))) + -log(e^(-ln)/(1+e^(-ln)))
= log(1+e^(-lp))(1+e^(ln)) # eq2

I think #eq1 and #eq2 are not same
